pkg/controllers/sync/version/manager.go

```diff
@@ -250,39 +255,33 @@ func (m *VersionManager) list(stopChan <-chan struct{}) (runtimeclient.ObjectLis
 // load processes a list of versions into in-memory cache.  Since the
 // version manager should not be used in advance of HasSynced
 // returning true, locking is assumed to be unnecessary.
-func (m *VersionManager) load(versionList runtimeclient.ObjectList, stopChan <-chan struct{}) bool {
-	typePrefix := PropagatedVersionPrefix(m.targetKind)
+func (m *VersionManager) load(ctx context.Context, versionList runtimeclient.ObjectList) bool {
 	objs, err := meta.ExtractList(versionList)
 	if err != nil {
 		return false
 	}
 	for _, obj := range objs {
 		select {
-		case <-stopChan:
-			m.logger.V(4).Info("Halting version manager load due to closed stop channel")
+		case <-ctx.Done():
+			m.logger.Info("Halting version manager load due to closed stop channel")
 			return false
 		default:
 		}
 
 		qualifiedName := common.NewQualifiedName(obj)
-		// Ignore propagated version for other types
-		if strings.HasPrefix(qualifiedName.Name, typePrefix) {
-			m.versions[qualifiedName.String()] = obj.(runtimeclient.Object)
-		}
+		m.versions[qualifiedName.String()] = obj.(runtimeclient.Object)
 	}
 	m.Lock()
 	m.hasSynced = true
 	m.Unlock()
-	m.logger.WithValues("federated-kind", m.federatedKind).
-		V(4).Info("Version manager for federatedKind synced")
+	m.logger.Info("Version manager synced")
 	return true
 }
 
 // versionQualifiedName derives the qualified name of a version
-// resource from the qualified name of a template or target resource.
+// resource from the qualified name of a federated object.
 func (m *VersionManager) versionQualifiedName(qualifiedName common.QualifiedName) common.QualifiedName {
```

Comment:
Is there still a need for this method now that we have declared in the API definition that the name of the `PropagatedVersion` and `ClusterPropagatedVersion` is identical to the name of a `FederatedObject` and `ClusterFederatedObject` respectively?

Reply:
Technically no. Changing the implementation of this method was the least invasive way to change PropagatedVersion naming. I can remove it if it's an issue.